### ISSUE TITLE
SP5: `Plot.ScaleFactor` does not compensate for figure dimensions

### ISSUE THREAD

**Issue by swharden** [ScottPlot/ScottPlot#2770]
https://github.com/ScottPlot/ScottPlot/pull/2766#issuecomment-1641203326

and also the cookbook example doesn't do what I'd expect

I think the issue is that scaling somewhere needs to tell the plot that its size is different. I'll take a closer look shortly.

**Comment by DmitryKotenev** [comment]
I have found `Surface.Canvas.Scale(scaleFactor)` with scaleFactor > 1 leads to **decrease** dimensions of the Canvas. I suppose it's needed to right work of underlying layers (SkiaSharp etc.). In current instruction flow defenition the size of Canvas occurs before scaling, and it leads to actually render canvas with wrong (not scaled) dimentions.

I wrote commit to fork. As I change `public void Render(SKCanvas canvas, int width, int height)`, I don't confident, that it's correct for merging, but it determines problem, and may be helpfull: 9dac043c96d92a58a297721d928c74fe763b5a1a
